File: gyroTest/src/main/java/com/stulnikov/arduino/arduinoBt/gyrotest/gyrotest/dagger/PresenterComponent.java
package com.stulnikov.arduino.arduinoBt.gyrotest.gyrotest.dagger;

import com.stulnikov.arduino.arduinoBt.gyrotest.gyrotest.MainActivity;

import dagger.Component;

/**
 * @author alexeystulnikov 2/28/17.
 */
@Component(modules = {PresenterModule.class}, dependencies = ApplicationComponent.class)
@PerActivity
public interface PresenterComponent {

    void inject(MainActivity mainActivity);
}

File: gyroTest/src/main/kotlin/com/stulnikov/arduino/arduinoBt/gyrotest/gyrotest/dagger/PresenterComponent.kt
package com.stulnikov.arduino.arduinoBt.gyrotest.gyrotest.dagger

import com.stulnikov.arduino.arduinoBt.gyrotest.gyrotest.MainActivity
import dagger.Component

/**
 * @author alexeystulnikov 2/28/17.
 */
@Component(modules = [
    PresenterModule::class,
    SensorModule::class
], dependencies = [
    ApplicationComponent::class
])
@PerActivity
interface PresenterComponent {
    fun inject(mainActivity: MainActivity)
}